speedy-recovery-client/src/components/secured/shared/Header.js

```diff
@@ -78,6 +90,47 @@ class Header extends Component {
       </header>
     );
   }
+
+  componentDidMount() {
+    this._isMounted = true;
+    if (this.props.role === "Practitioner"){
+      const id = this.props.practitionerid;
+      this.setState({ id });
+    }
+    if (this.props.role === "Parent"){
+      const id = this.props.userid;
+      this.setState({ id });
+
+    }
+    this.setMessageNum();
+    this.timer = setInterval(() => {
+      this.setMessageNum();
+    }, 5000);
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+    this.timer && clearTimeout(this.timer);
+  }
+
+  setMessageNum = () => {
+    if (this.props.role !== "Patient") {
+      getConversation(this.state.id)
```

Comment:
@fabihaahmed Didn't you say you'd lift the state to the closest common ancestor (probably App.js), so we don't have multiple components calling `getConversations` and `getMessages`, but just one? And then hand down state to other components?